app/filters/good_job/jobs_filter.rb

```diff
@@ -25,7 +25,14 @@ def filtered_query(filter_params = params)
 
       query = query.job_class(filter_params[:job_class]) if filter_params[:job_class].present?
       query = query.where(queue_name: filter_params[:queue_name]) if filter_params[:queue_name].present?
-      query = query.search_text(filter_params[:query]) if filter_params[:query].present?
+      query = case filter_params[:query].presence
+              when /\A[0-9a-f]{8}(?:-[0-9a-f]{4}){3}-[0-9a-f]{12}\z/i
+                query.where(active_job_id: filter_params[:query])
+              when nil
+                query # no query parameter were specified
+              else
+                query.search_text(filter_params[:query])
+              end
```

Comment:
This would break anything in which a UUID is in arguments or error or if they want to search by GoodJob's UUID rather than Active Job.

I sorta feel like here should be a fast-path where it's like:

```
  if looks_like_uuid
    uuid_query = query.where(id: uuid).or query.where(active_job_id: uuid)
  end

  if uuid_query&.any?
    query = uuid_query
  else
    # more complex query
  end
```